sys/net/gnrc/application_layer/dhcpv6/client.c

```diff
@@ -174,6 +174,17 @@ void dhcpv6_client_conf_prefix(unsigned iface, const ipv6_addr_t *pfx,
             gnrc_rpl_root_init(CONFIG_GNRC_RPL_DEFAULT_INSTANCE, &addr, false, false);
         }
     }
+
+    /* start advertising subnet */
+    gnrc_ipv6_nib_change_rtr_adv_iface(netif, true);
+}
+
+void dhcpv6_client_conf_done(unsigned iface)
+{
+    gnrc_netif_t *netif = gnrc_netif_get_by_pid(iface);
+
+    /* inform upstream about subnets */
```

Comment:
As someone reading this code after the fact and not knowing what the hell you are doing, this comment might be a bit confusing. "What subnets?" "I only have my RPL-DODAG based on the DHCPv6-advertised prefix, I don't want any more subnets advertised to the upstream... can I deactivate this?"

Also maybe don't enforce your use case on others ;-).

Reply:
But the border router has it's own subnet for the 6LoWPAN too.
If the upstream router manages a /56 and hands the border router a /62 for use in the WPAN, nodes in the upstream network still have to route via the border router if the destination is in the /62 subnet, they must not assume the host is on-link.

Reply:
I think my main problem with this change is that it is separate from the ability to handle the route information option to me. Can we move the behavior changes for DHCPv6 to a separate PR?

Reply:
sure!
I just wanted to present the big picture to get the confirmation that this is the right thing to do.